File: Gestor Hospitalario/Models/Especialidad.cs
using System.ComponentModel.DataAnnotations;

namespace Gestor_Hospitalario.Models
{
    public class Especialidad
    {
        [Key]
        public int EspecialidadID { get; set; }
        public string Nombre { get; set; } = string.Empty;

        public virtual ICollection<Medico> Medicos { get; set; } = new List<Medico>();
    }
}

File: gestor-hospitalario-react/src/@types/especialidad.ts
export interface Especialidad {
  especialidadID: number;
  nombre: string;
  descripcion: string;
}

export interface EspecialidadCreate {
  nombre: string;
  descripcion: string;
}

export interface EspecialidadUpdate extends EspecialidadCreate {
  especialidadID: number;
}
